### ISSUE TITLE
pass: correctness issues

### ISSUE THREAD

**Issue by zx2c4** [NixOS/nixpkgs#23223]
From the .nix:

>    installFlags = [ "PREFIX=$(out)" ];

You probably should add "WITH_ALLCOMP=yes BASHCOMPDIR=...", filling in each of the COMPDIRs with the paths that Nix uses (which are mentioned earlier in the same .nix file). See https://git.zx2c4.com/password-store/tree/INSTALL for more details. Alternatively, see what the homebrew formula does -- https://github.com/zx2c4/homebrew-core/blob/fb83d4a39f5bef83f1f5cda047ff40b5ede1f965/Formula/pass.rb#L22

Finally, are you sure `PREFIX=$(out)` is what you want? I don't know how .nix usually does things, but I imagine you actually want `DESTDIR=$(out)` and `PREFIX={something else}`. It should follow the usual pattern of Makefile-based installations.

@fpletz

**Comment by Mic92** [comment]
`PREFIX=$(out)` or similar is what we use for all packages. `$out` will refer to the package path `/nix/store/<hash>-<packagename>-<version>`. Unlike other package manager, which files install to a temporary location,  nix installs it directly to the nix store.

**Comment by fpletz** [comment]
Thanks for the hints about the completions. Fixed that with just `WITH_ALLCOMP=yes` because the default compdirs are the ones we want. The previous maintainers created them manually so that the Makefile would detect those.

Actually, we do want to override `PREFIX` because we're installing pass into a path like `/nix/store/nggxhsrlww2m4rny3dvjmlh63z9fx88x-password-store-1.7` and all dependencies are in similar prefixes. We don't have anything useful in `/usr` or any other common prefix. We're breaking the FHS intentionally to gain purity. That's also the reason we need a wrapper script to make all the needed executables available to pass.

I'm not yet sure how we will handle extensions, though, so I'll keep this issue open until we handled those. I haven't had a chance to look at that.
